analytics/upress_reporting/helpers/download_data_aggregator.py

```diff
@@ -103,36 +104,83 @@ def _redact_s3_path(self, path):
             match_time = re.search(TIMESTAMP_REGEX, log_object)
             match_file_id = re.search(FILE_ID_REGEX, log_object)
             link_group = match_file_id.group(1)
-            title_parse = ""
-            id_parse = None
+
+            '''
+            To go from the log event data, we need to go from the link to the 
+            item, to the edition, to the work, and to the original source record.
+
+            Links are linked to items by the item_id. 
+            Items are linked to editions by the edition_id. 
+            Editions are linked to works by the work_id.
+            Editions are linked to records by the dcdw_uuids.
+            '''
 
             for item in self.db_manager.session.query(Item).filter(
-                Item.source == self.publisher
-            ):
+                    Item.source == self.publisher):
                 for link in (
                     self.db_manager.session.query(Link)
                     .join(ITEM_LINKS)
                     .filter(ITEM_LINKS.c.item_id == item.id)
                     .filter(Link.media_type == "application/pdf")
                     .filter(Link.url.contains(link_group.strip()))
-                    .all()
-                ):
-                    item_edit_id = item.edition_id
-                    for edit in self.db_manager.session.query(Edition).filter(
-                        Edition.id == item_edit_id
-                    ):
-                        title_parse = edit.title
-                        id_parse = edit.id
-
-            return [title_parse, match_time.group(0), id_parse]
+                        .all()):
+                    for edition in self.db_manager.session.query(Edition).filter(
+                            Edition.id == item.edition_id):
+                        title = edition.title
+                        copyright_year = self._pull_copyright_year(edition)
+
+                        for record in self.db_manager.session.query(Record).filter(
+                                Record.uuid.in_(edition.dcdw_uuids)):
+                            book_id = (record.source_id).split("|")[0]
+                            usage_type = self._determine_usage(record)
+                            isbns = [identifier.split(
+                                "|")[0] for identifier in record.identifiers if "isbn" in identifier]
+                            eisbns = [identifier.split(
+                                "|")[0] for identifier in record.identifiers if "eisbn" in identifier]
+
+                        for work in self.db_manager.session.query(Work).filter(
+                                Work.id == edition.work_id):
+                            authors = [author["name"]
+                                       for author in work.authors]
+                            disciplines = [subject["heading"]
+                                           for subject in work.subjects]
```

Comment:
This whole block is super nested and looks a little messy, would love any thoughts on how to refactor this and make it look cleaner!

Reply:
good callout - definitely agree. let's add a todo but we may want to simplify the query if possible.